index.js

```diff
@@ -27,45 +27,68 @@ exports.init = function (sbot, config) {
     return msg.value
   }
 
-  sbot.resolveIndexFeed = function resolveIndexFeed(feedId) {
+  sbot.resolveIndexFeed = function resolveIndexFeed(indexFeedId) {
     // we assume that if we have the feed, that we also have the meta
     // feed this index is a part of
 
     return pull(
-      pull.values([feedId]),
-      pull.asyncMap((feedId, cb) => {
-        if (!ref.isFeed(feedId))
-          return cb(new Error('Invalid feed ID: ' + feedId))
-
-        sbot.metafeeds.query.getMetadata(feedId, cb)
+      pull.values([indexFeedId]),
+      pull.asyncMap(function validateFeedId(feedId, cb) {
+        if (!ref.isFeed(feedId)) {
+          cb(new Error('Invalid feed ID: ' + feedId))
+        } else {
+          cb(null, feedId)
+        }
+      }),
+      pull.asyncMap(function getRootMetafeed(feedId, cb) {
+        sbot.metafeeds.findOrCreate(cb)
+      }),
+      pull.asyncMap(function getIndexesMetafeed(rootMF, cb) {
+        sbot.metafeeds.find(rootMF, (f) => f.feedpurpose === 'indexes', cb)
       }),
-      pull.asyncMap((content, cb) => {
-        if (!content || content.feedpurpose !== 'index' || !content.query)
-          return cb(new Error('Not a proper index feed: ' + feedId))
-        if (
-          content.querylang !== 'ssb-ql-0' &&
-          content.querylang !== 'ssb-ql-1'
-        ) {
-          return cb(new Error('Unknown querylang: ' + content.querylang))
+      pull.asyncMap(function getIndexFeed(indexesMF, cb) {
+        if (!indexesMF) return cb(null, null)
```

Comment:
should this do an error here?

Reply:
Ah, so this `null` value propagates below and will be treated as "Index not found". I thought that it would be useful if this `resolveIndexFeed` would *not* attempt to create the indexesMF nor the indexFeed, so I used `find` not `findOrCreate`

Reply:
Ah right. Makes sense not to create these